recordset/recordset.controller.js

```diff
@@ -114,40 +111,68 @@
+                elements.initialInnerHeight = $document[0].getElementsByClassName('main-body')[0].offsetHeight;
+                // get the footer
+                elements.footer = $document[0].getElementsByTagName('footer')[0];
+            } catch(error) {
+                $log.warn(error)
+            }
+            return elements;
+        }
+
+        function setFooterStyle() {
+            var elements = fetchBodyElements();
+
+            UiUtils.setFooterStyle(elements);
+        }
+
         $scope.$watch(function() {
             return recordsetModel.hasLoaded && recordsetModel.initialized;
         }, function (newValue, oldValue) {
             if (newValue) {
-                try {
-                    var elements = fetchMainElements();
-                    // if these 2 values are not set yet, don't set the height
-                    if(elements.navbarHeight && elements.bookmarkHeight) {
-                        UiUtils.setDisplayHeight(elements);
-                        // no need to fetch and verify the faceting elements (navbar and bookmark are the same container as the ones used in main elements function)
-                        if (chaiseConfig.showFaceting) UiUtils.setDisplayHeight(fetchFacetingElements());
-                    }
-                } catch(exp) {
-                    // fail silently
-                }
+                setRecordsetHeight();
+            }
+        });
+
+        // watch for the main body size to change
+        $scope.$watch(function() {
+            if (mainBodyEl && mainBodyEl[0]) {
+                return mainBodyEl && mainBodyEl[0].offsetHeight;
+            }
+        }, function (newValue, oldValue) {
+            if (newValue) {
+                setFooterStyle();
             }
         });
 
+
         angular.element($window).bind('resize', function(){
-            try {
-                if (recordsetModel.hasLoaded && recordsetModel.initialized ) {
-                    var elements = fetchMainElements();
-                    // if these 2 values are not set yet, don't set the height
-                    if(elements.navbarHeight && elements.bookmarkHeight) {
-                        UiUtils.setDisplayHeight(elements);
-                        // no need to fetch and verify the faceting elements (navbar and bookmark are the same container as the ones used in main elements function)
-                        if (chaiseConfig.showFaceting) UiUtils.setDisplayHeight(fetchFacetingElements());
-                    }
-                    $scope.$digest();
-                }
-            } catch(exp) {
-                // fail silently
+            if (recordsetModel.hasLoaded && recordsetModel.initialized ) {
+                setRecordsetHeight();
+                setFooterStyle();
+                $scope.$digest();
             }
         });
 
+        $timeout(function () {
+            mainBodyEl = $document[0].getElementsByClassName('main-body');
```

Comment:
Was there a reason behind the `$timeout` in this app? I feel like we should unify the way we're getting this element in all three apps. It just makes it easier to follow the code, and eventually we can create a common service for all these common logics.